fiftyone/server/mutation.py

```diff
@@ -178,31 +179,19 @@ async def set_view(
         state.selected_labels = []
 
         result_view = None
-        ds = fod.load_dataset(dataset_name)
-        view_name = None
         if saved_view_slug is not None:
             try:
+                # Load a DatasetView using a slug
+                ds = fod.load_dataset(dataset_name)
                 doc = ds._get_saved_view_doc(saved_view_slug, slug=True)
-                view_name = doc.name
+                result_view = ds._load_saved_view_from_doc(doc)
             except:
-                pass
-
-        if view_name is not None:
-            # Load a saved view by name
-
-            if ds.has_saved_view(view_name):
-                # Load a saved dataset view by name
-                result_view = ds.load_saved_view(view_name)
-                view = result_view._serialize()  # serialized view stages
-                # Set view state
-                state.view = result_view
-                state.view_name = result_view.name
-                state.saved_view_slug = saved_view_slug
-        else:
-            state.view_name = None
-            state.saved_view_slug = None
+                logging.error(
+                    f"[mutation.py] ERROR: Unable to load saved view with "
+                    f"slug={saved_view_slug}"
+                )
 
-        if result_view is None:
+        if not result_view:
```

Comment:
This will call [`__bool__`](https://github.com/voxel51/fiftyone/blob/7dd227cc52fe5000067c92363d57dca2584f76bd/fiftyone/core/collections.py#L181) which will in turn call `__len__` which is very bad. `is None` is best

Reply:
oh good to know! For other things, sometimes the None specific check lets "" and [] get by, which can trigger errors downstream. However, it's safe here to use None because result_view is declared None at the beginning.